### PR TITLE
Test and fix custom labels support in mixin

### PR DESCRIPTION
**Pull request by pracucci**
#### What this PR does
We just merged #1651 which adds support to customise the cluster label. We already had config options to customise the instance (eg. pod) and node labels.

However, are we relatively confident the labels customization is applied everywhere? It's very difficult to spot during code reviews, so I though if we could add any test. In this PR I'm proposing to introduce a basic test (in bash) checking if, after renaming cluster/instance/node labels, they're still used somewhere.

Thanks to these tests, I've spot several issues that I'm fixing in this PR too. In theory this PR should be a noop when mixin is compiled with default config, but you will see diff in `operations/mimir-mixin-compiled/alerts.yaml`. Reason is that fix these issues I changed the usage of hardcoded `instance` label with the configured one which is `pod`. At Grafana Labs, `instance` and `pod` labels can be used interchangeably, because they both identify the pod.

#### Which issue(s) this PR fixes or relates to
N/A

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: operations/mimir-mixin/alerts.libsonnet [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
   prometheusAlerts+::
+    { _config:: $._config + $._group_config } +
```
Comment (pracucci) [inline operations/mimir-mixin/alerts.libsonnet:3]: Moved here so I can extend `_config` in `alert-utils.libsonnet`.

File: operations/mimir-mixin-compiled/alerts.yaml [hunk 2]
```diff
@@ -476,7 +476,7 @@ groups:
       # Only if the ingester has ingested samples over the last 4h.
       (max by(cluster, namespace, pod) (max_over_time(cluster_namespace_pod:cortex_ingester_ingested_samples_total:rate1m[4h])) > 0)
       and
-      # Only if the ingester was ingesting samples 4h ago. This protects from the case the ingester instance
+      # Only if the ingester was ingesting samples 4h ago. This protects from the case the ingester replica
```
Comment (osg-grafana) [inline operations/mimir-mixin-compiled/alerts.yaml:479]:
```suggestion
      # Only if the ingester was ingesting samples 4h ago. This protects against cases where the ingester replica
```
```suggestion
      # Only if the ingester was ingesting samples 4h ago. This protects from the case the ingester replica
```

Comment (osg-grafana) [reply operations/mimir-mixin-compiled/alerts.yaml:479]: nit

Comment (osg-grafana) [reply operations/mimir-mixin-compiled/alerts.yaml:479]: nit

File: operations/mimir-mixin-tests/test-custom-labels-asserts.sh [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  fi
+
+  # Check rules.
+  echo "Checking ${RULES_FILE}"
+  MATCHES=$(yq eval '.. | select(.expr?).expr | sub("\n", " ")' "${RULES_FILE}" | grep -E "${QUERY_REGEX}" || true)
+  if [ -n "$MATCHES" ]; then
+    assert_failed "The rules at ${RULES_FILE} contains unexpected references to '${LABEL}' label in some queries:\n$MATCHES"
+  fi
+
+  MATCHES=$(yq eval '.. | select(.record?).record | sub("\n", " ")' "${RULES_FILE}" | grep -Eo '^[^:]+' | grep -E "${RULE_PREFIX_REGEX}" || true)
```
Comment (replay) [inline operations/mimir-mixin-tests/test-custom-labels-asserts.sh:59]: what is the purpose of the `grep -Eo '^[^:]+'` part?

**Review by replay — APPROVED** [review]
This looks great, thanks! I just added some comments of minor importance

File: operations/mimir-mixin-tests/test-custom-labels-asserts.sh (hunk 3 above)
Comment (pracucci) [reply operations/mimir-mixin-tests/test-custom-labels-asserts.sh:59]: It's just to run the check in the first part of the recording rule, where we have the grouping labels. For example, if `pod` is a grouping label then it's bad, but if `pod` is part of the metric name then it's OK (eg. `kube_pod_.*`).

Comment (replay) [reply operations/mimir-mixin-tests/test-custom-labels-asserts.sh:59]: ic, makes sense, thx for explaining

**Comment by pracucci** [comment]
_I will wait next week before merging, both to give the opportunity to other engineers to review it and not deploy these alerts changes right before Easter._

**Comment by pracucci** [comment]
I double checked the compiled alerts, manually tested few alerts and changes LGTM. Going to merge it.

**Pull request merged**